src/main/java/fi/aalto/cs/apluscourses/model/task/Task.java

```diff
@@ -96,5 +105,24 @@ public synchronized boolean startTask(ActivityFactory activityFactory) {
         : JsonUtil.parseObject(jsonObject, JSONObject::get,
             Function.identity(), Function.identity())::get;
   }
+
+  private void startTimer() {
+    timer.scheduleAtFixedRate(new TaskRefresher(), REFRESH_INTERVAL, REFRESH_INTERVAL);
+  }
+
+  private class TaskRefresher extends TimerTask {
+
+    @Override
+    public void run() {
+      ApplicationManager.getApplication().invokeLater(() -> {
+        if (presenter == null) {
+          return;
+        }
+        if (!presenter.isVisible()) {
+          presenter.highlight();
+        }
+      });
```

Comment:
Here we have at least a couple of threading issues, I'm afraid...
1) `presenter` can become null after the null-check so line 121 can throw a null pointer exception.
2) `presenter.highlight()` can also be called after `endTask` has executed `presenter.removeHighlight()`, leaving the highlight on when the task is done. (NOTE: Changing `endTask` in a way that it sets `presenter` null first and, only after that, calls `removeHighlight` with a locally stored reference WON'T fix this!)

All in all, I find this design kind of error-prone, and difficult to reason about. I strongly suggest that we move refreshing logic (timer) inside the presenters. If I have understood the issue correctly, the fact that highlighting can suddenly just disappear is due to some weird behaviour of IntelliJ/Swing. So, why would we want that anomaly "leak" onto our `Presenter` interface? IMO, it would make much more sense if `Task` could use presenters expecting them to work as they should be (without spurious disappearances) and they are presenter objects who make sure this is fulfilled.